### PR TITLE
Add Apache Spark 3.1.4-SNAPSHOT Shims

### PR DESCRIPTION
**Pull request by nartal1**
This fixes https://github.com/NVIDIA/spark-rapids/issues/4789.

This PR adds Spark-3.1.4-SNAPSHOT shims. Added test to verify that it works as expected.

### CODE REVIEW

**Comment by nartal1** [comment]
@tgravescs Could you please let me know the priority of this over https://github.com/NVIDIA/spark-rapids/pull/4857. I can wait until that PR is merged and upmerge my changes to reflect the new design. If we want to enable 3.1.4-SNAPSHOT nightly builds sooner then I think this should be merged earlier.

**Comment by tgravescs** [comment]
either way, I think 3.1.4 can wait if the other PR is large

**Comment by nartal1** [comment]
> either way, I think 3.1.4 can wait if the other PR is large

Thanks for the confirmation. Have marked this PR as draft. Will upmerge once other PR is merged.

**Comment by razajafri** [comment]
Thanks @nartal1

**Comment by tgravescs** [comment]
@nartal1 please update, the other pr has been merged

File: docs/additional-functionality/rapids-shuffle.md
Comment (tgravescs) [inline docs/additional-functionality/rapids-shuffle.md:297]: leave this off until it becomes not a snapshot

File: jenkins/Jenkinsfile-blossom.premerge
Comment (tgravescs) [inline jenkins/Jenkinsfile-blossom.premerge:331]: this directory doesn't exist anymore, revert

Comment (nartal1) [reply jenkins/Jenkinsfile-blossom.premerge]: done

File: sql-plugin/src/test/314/scala/com/nvidia/spark/rapids/shims/spark314/Spark314ShimsSuite.scala
Comment (nartal1) [reply sql-plugin/src/test/314/scala/com/nvidia/spark/rapids/shims/spark314/Spark314ShimsSuite.scala]: removed these tests.

**Comment by tgravescs** [comment]
build

**Review by tgravescs — APPROVED** [review]
We need to get integration test pipeline going for this.

**Pull request merged**
